src/app/players/pdf/pdf.component.ts

```diff
@@ -19,28 +20,36 @@ export class PdfComponent implements OnInit {
   @ViewChild('preview', { static: false }) previewElement: ElementRef;
   public queryParams: any;
   public contentDetails: any;
-  playerConfig = this.configService.playerConfig.PDF_PLAYER;
+  playerConfig: any;
```

Comment:
Why dont you have base player component & move the common logic & properties into base component. Then each PDF, epub, video etc.. will write specific logic of each player.